### PR TITLE
Support libavif v0.60 and set proper color space as possible.

### PR DESCRIPTION
**Pull request by ledyba-z**
Hi.

In this PR, set proper color profile values to CGImage (if possible).

This PR and https://github.com/AOMediaCodec/libavif/pull/98 may resolve https://github.com/SDWebImage/SDWebImageAVIFCoder/issues/12 .

Please take a look!

### CODE REVIEW

File: SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m [hunk 1]
```diff
@@ -17,6 +17,510 @@ static void FreeImageData(void *info, const void *data, size_t size) {
...
+        *ref = defaultColorSpace;
+        *shouldRelease = FALSE;
+        return;
+    }
+    if(avif->profileFormat == AVIF_PROFILE_FORMAT_ICC) {
+        if(avif->icc.data && avif->icc.size) {
+            if(@available(macOS 10.12, iOS 10.0, tvOS 10.0, *)) {
+                *ref = CGColorSpaceCreateWithICCData(avif->icc.data);
+                *shouldRelease = TRUE;
+                return;
```
Comment (dreampiggy) [inline SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m]: call CGColorSpaceCreateWithICCProfile for iOS 8~9

File: SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m [hunk 2]
```diff
@@ -17,6 +17,510 @@ static void FreeImageData(void *info, const void *data, size_t size) {
...
+            } else {
+                p3linear = defaultColorSpace;
+            }
+        });
+        *ref = p3linear;
+        *shouldRelease = FALSE;
+        return;
+    }
+
+    *ref = CreateColorSpaceRGB(colorPrimaries, transferCharacteristics);
```
Comment (dreampiggy) [inline SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m]: This is the fallback case for CGColorSpace which does not supported ?

File: SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m (hunk 1 above)
Comment (ledyba-z) [reply SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m]: OK. I called it.

File: SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m (hunk 2 above)
Comment (ledyba-z) [reply SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m]: This branch is for when there is no corresponding built-in color space in `CGColorSpace.h`, like `kCGColorSpaceSRGB`.
If there is better way to create color spaces, please tell me.

Comment (dreampiggy) [reply SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m]: OK. Seems we need to refactory soon....There are so many calc code and utils C function right now :). Don't need to do this right now.

Comment (ledyba-z) [reply SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m]: OK. I removed the code that creates CGColorSpace using vImage, and left a TODO comment about it.

Yeah, I also felt we need refactoring.

**Comment by dreampiggy** [comment]
😅 Seems you still trying some changes. If you find it's ready to merge, @ me again.

And, can CI test image been passed after this color space support changes ?

File: SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m (hunk 2 above)
Comment (ledyba-z) [reply SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m]: Sorry, I mis-understand your message. I reverted https://github.com/SDWebImage/SDWebImageAVIFCoder/pull/13/commits/71ecfa1d0a2273d52f9042199221708297da319a .

**Comment by ledyba-z** [comment]
@dreampiggy Sorry! I've done!

> And, can CI test image been passed after this color space support changes ?

Unfortunatelly, https://github.com/AOMediaCodec/libavif/pull/98 should also be merged to libavif, and we have to wait the new version release of libavif to fix CI tests.

**Comment by ledyba-z** [comment]
@dreampiggy Hi! Now libavif v0.6.0 is supported, and all end-to-end tests (Github Actions) have finally passed!
Could you take a look?

**Comment by ledyba-z** [comment]
Thank you!

**Comment by ledyba-z** [comment]
I tested the new alpha channel codes with [plum-blossom-large.profile0.8bpc.yuv420.avif](https://github.com/link-u/avif-sample-images/blob/da87022ba412d77f53e80ec7e4f26064076f51f5/plum-blossom-large.profile0.8bpc.yuv420.avif)

## iPhone 6

```
v0.5.2 : 2.0247561931610107 sec
master: 1.7678682804107666 sec
```
It runs faster about 13%

## iPhone 8

```
v0.5.2: 0.7256550788879395 sec
master: 0.5414080619812012 sec
```
It runs faster about 25%

**Pull request merged**